### PR TITLE
gh-98831: rewrite CHECK_EG_MATCH opcode in the instruction definition DSL

### PR DESCRIPTION
**Pull request by iritkatriel**
The change in exception_group_match is needed because previously in the case of match == None the opcode didn't look at rest, but now it pushes to the stack (even though it's already there) to keep the code uniform with the match != None case.

* Issue: gh-98831

### CODE REVIEW

**Review by gvanrossum — APPROVED** [review]
Looks like this one wasn't trivial. Thanks!

File: Misc/NEWS.d/next/Core and Builtins/2023-01-23-22-06-07.gh-issue-98831.FkCDuV.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Rewrite the implementation of :opcode:`CHECK_EG_MATCH` in the instruction
+definition DSL.
```
Comment (gvanrossum) [inline Misc/NEWS.d/next/Core and Builtins/2023-01-23-22-06-07.gh-issue-98831.FkCDuV.rst]: Actually I've not been adding news for anything related to the generator, assuming nobody cares about its progress.

**Comment by iritkatriel** [comment]
I'll kick off the buildbots in case refcount.

**Comment by gvanrossum** [comment]
PPC64 Fedora PR is failing with "No space left on device", so I wouldn't worry about that one. :)

**Pull request merged**